vault-github/run.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+VAULT_ADDR=${VAULT_ADDR}
+VERSION=${VERSION}
+AUTH_PATH=${AUTH_PATH}
+GITHUB_EXTERNAL_AUTH_ID=${GITHUB_EXTERNAL_AUTH_ID}
+
+# Ensure curl is installed
+if ! command -v curl &>/dev/null; then
+    printf "curl is not installed. Please install curl in your image.\n"
+    exit 1
+fi
+
+# Fetch latest version of Vault if VERSION is 'latest'
+if [ "${VERSION}" = "latest" ]; then
+    LATEST_VERSION=$(curl -s https://releases.hashicorp.com/vault/ | grep -oP 'vault/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
+    if [ -z "$LATEST_VERSION" ]; then
+        printf "Failed to determine the latest Vault version.\n"
+        exit 1
+    fi
+    VERSION=$LATEST_VERSION
+fi
+
+# Check if vault is installed and has the correct version
+installation_needed=1
+if command -v vault &>/dev/null; then
+    CURRENT_VERSION=$(vault version | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+    if [ "$CURRENT_VERSION" = "$VERSION" ]; then
+        printf "Vault version $CURRENT_VERSION is already installed and up-to-date.\n\n"
+        installation_needed=0
+    fi
+fi
+
+if [ $installation_needed -eq 1 ]; then
+    # Check if unzip or busybox is installed
+    if ! command -v unzip &>/dev/null; then
+        if command -v busybox &>/dev/null && busybox --list | grep -q '^unzip$'; then
+            alias unzip='busybox unzip'
+            printf "Using busybox unzip.\n"
+        else
+            printf "unzip is not installed and busybox unzip is not available. Please install unzip in your image.\n"
+            exit 1
+        fi
+    fi
```

Comment:
I think the alias won't work here, and pretty sure not in POSIX shells (sh).

How about defining the following functions instead?

```sh
fetch() {
    dest="$1"
    url="$2"
    if command -v curl; then
        command curl -sSL --fail "${url}" -o "${dest}"
    elif command -v wget; then
        command wget -O "${dest}" "${url}"
    elif command -v busybox; then
        busybox wget -O "${dest}" "${url}"
    else
        printf "curl, wget, or busybox is not installed. Please install curl or wget in your image.\n"
        exit 1
    fi
}

unzip() {
    if command -v unzip; then
        command unzip "$@"
    elif command -v busybox; then
        busybox unzip "$@"
    else
        printf "unzip or busybox is not installed. Please install unzip in your image.\n"
        exit 1
    fi
}
```

Then it can be used as `fetch vault.zip "https://..."` and `unzip -o vault.zip` and keep the "main code" clean. It'll also take care of any alias problems.